src/useFetchye.js

```diff
@@ -14,48 +14,65 @@
  * permissions and limitations under the License.
  */
 
-import { useContext, useEffect, useRef } from 'react';
+import { useEffect, useRef } from 'react';
 import { runAsync } from './runAsync';
-import { defaultFetcher } from './defaultFetcher';
 import { computeKey } from './computeKey';
-import { isLoading } from './isLoading';
-import { FetchyeContext } from './FetchyeContext';
-import { getCacheByKey } from './getCacheByKey';
+import {
+  isLoading,
+} from './queryHelpers';
+import { useFetchyeContext } from './useFetchyeContext';
 import { defaultMapOptionsToKey } from './defaultMapOptionsToKey';
 
+const passInitialData = (value, initialValue, numOfRenders) => (numOfRenders === 1
+  ? value || initialValue
+  : value);
+
 export const useFetchye = (
   key,
   { mapOptionsToKey = (options) => options, ...options } = { },
-  fetcher = defaultFetcher) => {
+  fetcher) => {
   const {
-    useFetchyeSelector, dispatch, fetchClient,
-  } = useContext(FetchyeContext);
-  const cache = useFetchyeSelector();
+    defaultFetcher, useFetchyeSelector, dispatch, fetchClient,
+  } = useFetchyeContext();
+  const selectedFetcher = typeof fetcher === 'function' ? fetcher : defaultFetcher;
   const computedKey = computeKey(key, defaultMapOptionsToKey(mapOptionsToKey(options)));
-  const { data, loading, error } = getCacheByKey(cache, computedKey);
-  const isFirstRender = useRef(!data);
+  const selectorState = useFetchyeSelector(computedKey.hash);
+  const numOfRenders = useRef(0);
+  numOfRenders.current += 1;
   useEffect(() => {
     if (options.lazy || !computedKey) {
```

Comment:
I think we briefly discussed about the name of this option to be something like `defer` instead of `lazy` here is the argument for the team to choose: 

"A defer statement defers the execution of a function until the surrounding function returns. The deferred call's arguments are evaluated immediately, but the function call is not executed until the surrounding function returns"

"Lazy loading (also known as asynchronous loading) is a design pattern commonly used in computer programming to defer initialization of an object until the point at which it is needed. It can contribute to efficiency in the program's operation if properly and appropriately used.

I thought if we are making public api changes this is the time

Reply:
According to internal feedback, we will go with `defer`.

Reply:
Fixed in https://github.com/americanexpress/fetchye/pull/9/commits/0ef5f8bb758a5802f1dcfbe1e71338b61c3f1159